test/IECoreMaya/LiveSceneTest.py

```diff
@@ -1008,6 +1008,40 @@ def testMultiCurves( self ) :
 		maya.cmds.currentTime( "0.0sec" )
 		mergedCurves = scene.readObject( 0 )
 		self.assertEqual( mergedCurves.numCurves(), 2 )
+	
+	def testMultiCurvesWithDifferentDegrees( self ) :
+
+		maya.cmds.createNode( "transform", name="sharedParent" )
+
+		maya.cmds.curve( d=3, p=[ ( 0, 0, 0 ), ( 1, 0, 0 ), ( 2, 0, 0 ), ( 3, 0, 0 ) ], k=( 0, 0, 0, 1, 1, 1 ), n="curve1" )
+		maya.cmds.curve( d=1, p=[ ( 0, 0, 0 ), ( 0, 0, 1 ) ], k=( 0, 1 ), n="curve2" )
+		maya.cmds.curve( d=1, p=[ ( 0, 0, 0 ), ( 0, 0, 1 ) ], k=( 0, 1 ), n="curve3" )
+
+		maya.cmds.select( "curveShape1", "curveShape2", "curveShape3", "sharedParent")
+		maya.cmds.parent( s=True, r=True )
+
+		scene = IECoreMaya.LiveScene()
+		scene = scene.child('sharedParent')
+		maya.cmds.currentTime( "0.0sec" )
+		curve = scene.readObject( 0 )
+		self.assertEqual( curve.numCurves(), 1 ) # Still has object but curves are not merged.
```

Comment:
I guess we could be asserting the "curves are not merged" part by checking `curve["P"].data.size()`?